src/core/idb-cache.js

```diff
@@ -0,0 +1,109 @@
+/**
+ * A promise based interface to a key-value based indexedDB cache
+ */
+
+const databases = new Map();
+
+export default class IDBCache {
+  /**
+   * @constructor
+   * @param {string} name             name of database
+   * @param {array} stores            name of stores
+   * @param {object} Options
+   *    @param {integer} version      database version
+   *    @param {string} defaultStore  name of default store
+   */
+  constructor(name, stores = [], { version = 1, defaultStore }) {
```

Comment:
i think it should not be optional. it would be pretty useless if user didn't provide a store (didn't think about this earlier). So, I'll make this required

Reply:
Good point! Should we then make it mandatory and require 1?

Reply:
yes. atleast one must be required.
(unless we want to create another method like `addStore` - which adds unnecessary complexity)